NTupleMaker/bin/SynchNTupleProducer_2017.cpp

```diff
@@ -1916,4 +1976,44 @@ void FillGenTree(const AC1B * analysisTree, Synch17GenTree *gentree, TString ch)
     gen_acott(analysisTree,gentree,LeadingtauIndex,TrailingtauIndex);
 
 
+//here fill the generator vertices to have the information present in tree
+
+  for (unsigned int igen=0; igen<analysisTree->genparticles_count; ++igen) {
+    if (analysisTree->genparticles_pdgid[igen]==23||analysisTree->genparticles_pdgid[igen]==24||
+	analysisTree->genparticles_pdgid[igen]==25||analysisTree->genparticles_pdgid[igen]==35||analysisTree->genparticles_pdgid[igen]==36) {
+      gentree->GenVertexX=analysisTree->genparticles_vx[igen];
```

Comment:
Please change the condition on the genparticles to match the following one:
(abs(analysisTree->genparticles_pdgid[ipart])==25||
abs(analysisTree->genparticles_pdgid[ipart])==35||
abs(analysisTree->genparticles_pdgid[ipart])==36)&&
 analysisTree->genparticles_isLastCopy[ipart]==1

Reply:
Yes, adding W and Z make sense. Back when the code was originally planned the generator level tree was organized so that it would contain only taus coming from a Higgs.
Given that now we want to use the gen.level information for the PV adding W and Z bosons make sense.

As for the info variable, I'd rather not use that because it is taken directly from CMSSW and is harder to read in the code. I think the variable `genparticles_isPrompt` would be best in this case.